webapp/src/main/java/com/box/l10n/mojito/quartz/QuartzPollableTaskScheduler.java

```diff
@@ -161,6 +172,34 @@ public <I, O> PollableFuture<O> scheduleJob(QuartzJobInfo<I, O> quartzJobInfo) {
     return new QuartzPollableFutureTask<O>(pollableTask, jobOutputType);
   }
 
+  private <I, O> QuartzPollableFutureTask<O> getPendingPollableFutureIfExists(
+      QuartzJobInfo<I, O> quartzJobInfo, Scheduler scheduler) {
+    logger.debug(
+        "DisallowConcurrentExecution is set, checking if the job already has a trigger in a blocked state");
+    String jobKeyName = getKeyName(quartzJobInfo.getClazz(), quartzJobInfo.getUniqueId());
+    try {
+      List<Trigger> triggers =
+          (List<Trigger>) scheduler.getTriggersOfJob(new JobKey(jobKeyName, DYNAMIC_GROUP_NAME));
+      for (Trigger trigger : triggers) {
+        if (scheduler.getTriggerState(trigger.getKey()).equals(Trigger.TriggerState.BLOCKED)) {
+          // Should only ever be one trigger in this state when @DisallowConcurrentExecution is set
+          logger.debug(
+              "Trigger {} is in blocked state, retrieving associated pollable task",
+              trigger.getKey());
+          PollableTask pollableTask =
+              pollableTaskService.getPollableTask(
+                  Long.valueOf(
+                      trigger.getJobDataMap().getString(QuartzPollableJob.POLLABLE_TASK_ID)));
+          return new QuartzPollableFutureTask<>(pollableTask, getJobOutputType(quartzJobInfo));
+        }
+      }
+    } catch (SchedulerException e) {
+      logger.debug(
```

Comment:
is that expected to happen or is an actual error? If the later, let's log as an "error"? in both case could be useful to have "e" logged too